File: code_samples/r/web_development_rest_api.r
# Web Development: Rest Api
# AI/ML Training Sample

RestApi <- setRefClass(
    "RestApi",
    fields = list(data = "character"),
    methods = list(
        initialize = function() {
            data <<- ""
        },
        process = function(input) {
            data <<- input
        },
        getData = function() {
            return(data)
        },
        validate = function() {
            return(nchar(data) > 0)
        }
    )
)

# Example usage
instance <- RestApi$new()
instance$process("example")
cat("Data:", instance$getData(), "\n")
cat("Valid:", instance$validate(), "\n")

File: code_samples/elixir/web_development_rest_api.ex
# Web Development: Rest Api
# AI/ML Training Sample

defmodule RestApi do
  defstruct data: ""
  
  def new(), do: %RestApi{}
  
  def process(%RestApi{} = struct, input) do
    %{struct | data: input}
  end
  
  def get_data(%RestApi{data: data}), do: data
  
  def validate(%RestApi{data: data}) do
    String.length(data) > 0
  end
end

# Example usage
instance = RestApi.new()
updated = RestApi.process(instance, "example")
IO.puts("Data: " <> RestApi.get_data(updated))
IO.puts("Valid: " <> to_string(RestApi.validate(updated)))
